exercises/practice/state-of-tic-tac-toe/.meta/src/reference/java/StateOfTicTacToe.java

```diff
@@ -0,0 +1,111 @@
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+class StateOfTicTacToe {
+    public GameStates determineState(String[] board) {
+        int xWin = 0;
+        int oWin = 0;
+
+        List<String> columns = getColumns(board);
+        List<String> diagonals = getDiagonals(board);
+        
+        int xCount = count('X', board);
+        int oCount = count('O', board);
+
+        if (oCount > xCount) {
+            throw new IllegalArgumentException("Wrong turn order: O started");
+        }
+
+        if (xCount > oCount + 1) {
+            throw new IllegalArgumentException("Wrong turn order: X went twice");
+        }
+
+        List<String> allCombos = new ArrayList<>();
+        
+        allCombos.addAll(Arrays.asList(board));
+        allCombos.addAll(columns);
+        allCombos.addAll(diagonals);
+        
+        for (String row : allCombos) {
+            row = row.trim();
+
+            if (row.equals("XXX")) {
+                if (xWin > 1 || oWin > 0) {
```

Comment:
I'm trying to understand why this uses `xWin > 1` but the next block (if the row equals `OOO`) doesn't use `oWin > 1`. Why is the logic not the same for both blocks?

Reply:
The reason of this are this tests:
- ``testFinishedGameWhereXWonViaARowAndAColumnVictory``
- ``testFinishedGameWhereXWonViaTwoDiagonalVictories``

if we remove them we can apply the reformat that you mentioned below.
Because this tests have winCount == 2, but the win is achieve by 1 move

Imagine a board like this, is valid and is X turn, so achieves the win with that move but the count is going to be 2 in the end, because a row and a column match the criteria
````
   |  X  | X
---------------
X  |  O  | O
---------------  
X  |  O  | O
````

Reply:
Ah yes, that is an interesting case that I didn't consider. In that case the current implementation is great!